tensorflow_quantum/core/src/circuit_parser_qsim.cc

```diff
@@ -0,0 +1,373 @@
+                            const SymbolMap& param_map, float* result) {
+  // find arg_name in proto.
+  const auto arg_v = op.args().find(arg_name);
+  if (arg_v == op.args().end()) {
+    return Status(tensorflow::error::INVALID_ARGUMENT,
+                  "Could not find arg: " + arg_name + " in op.");
+  }
+  // find proto arg field.
+  const auto proto_arg = arg_v->second;
+  *result = proto_arg.arg_value().float_value();
+  if (!proto_arg.symbol().empty()) {
+    // find symbol value in param_map.
+    const auto iter = param_map.find(proto_arg.symbol());
+    if (iter == param_map.end()) {
+      return Status(
+          tensorflow::error::INVALID_ARGUMENT,
+          "Could not find symbol in parameter map: " + proto_arg.symbol());
+    }
+    *result = iter->second.second;
+  }
+  return Status::OK();
+}
+
+// series of fixed signature gate builders.
+// there is no need to error check for unparseable symbols
+// or proto args not being present. Those errors are caught
+// upstream.
+
+// single qubit gate Create(time, q0)
+inline Status SingleConstantGate(
+    const Operation& op, const SymbolMap& param_map,
+    const std::function<QsimGate(unsigned, unsigned)>& create_f,
+    const unsigned time, QsimCircuit* circuit) {
+  int q0;
+  bool unused = absl::SimpleAtoi(op.qubits(0).id(), &q0);
+  circuit->gates.push_back(create_f(time, q0));
+  return Status::OK();
+}
+
+// two qubit gate Create(time, q0, q1)
+inline Status TwoConstantGate(
+    const Operation& op, const SymbolMap& param_map,
+    const std::function<QsimGate(unsigned, unsigned, unsigned)>& create_f,
+    const unsigned time, QsimCircuit* circuit) {
+  int q0, q1;
+  bool unused = absl::SimpleAtoi(op.qubits(0).id(), &q0);
+  unused = absl::SimpleAtoi(op.qubits(1).id(), &q1);
+  circuit->gates.push_back(create_f(time, q0, q1));
+  return Status::OK();
+}
+
+// single qubit eigen -> Create(time, q0, exponent, global_shift)
+inline Status SingleEigenGate(
+    const Operation& op, const SymbolMap& param_map,
+    const std::function<QsimGate(unsigned, unsigned, float, float)>& create_f,
+    const unsigned time, QsimCircuit* circuit) {
+  int q0;
+  bool unused;
+  float exp, exp_s, gs;
+  Status u;
+  unused = absl::SimpleAtoi(op.qubits(0).id(), &q0);
+  u = ParseProtoArg(op, "exponent", param_map, &exp);
+  if (!u.ok()) {
+    return u;
+  }
+  u = ParseProtoArg(op, "exponent_scalar", param_map, &exp_s);
+  if (!u.ok()) {
+    return u;
+  }
+  u = ParseProtoArg(op, "global_shift", param_map, &gs);
+  if (!u.ok()) {
+    return u;
+  }
+
+  circuit->gates.push_back(create_f(time, q0, exp * exp_s, gs));
+  return Status::OK();
+}
+
+// two qubit eigen -> Create(time, q0, q1, exp, gs)
+inline Status TwoEigenGate(
+    const Operation& op, const SymbolMap& param_map,
+    const std::function<QsimGate(unsigned, unsigned, unsigned, float, float)>&
+        create_f,
+    const unsigned time, QsimCircuit* circuit) {
+  int q0, q1;
```

Comment:
`create_f()` has unsigned qubit indices, but they are `int`. could you make them `unsigned int` or whatever `unsigned` type that you intended?

Reply:
Fixed.